File: py/ai/h2o/sparkling/ml/models/H2OMOJOPipelineModel.py
from pyspark.ml.param import *
from pyspark.ml.util import _jvm
from pyspark.sql.column import Column

from ai.h2o.sparkling.Initializer import Initializer
from ai.h2o.sparkling.ml.models import H2OMOJOSettings
from ai.h2o.sparkling.ml.models.H2OMOJOModelBase import H2OMOJOModelBase


class H2OMOJOPipelineModel(H2OMOJOModelBase):

    @staticmethod
    def createFromMojo(pathToMojo, settings=H2OMOJOSettings.default()):
        # We need to make sure that Sparkling Water classes are available on the Spark driver and executor paths
        Initializer.load_sparkling_jar()
        javaModel = _jvm().ai.h2o.sparkling.ml.models.H2OMOJOPipelineModel.createFromMojo(pathToMojo,
                                                                                                  settings.toJavaObject())
        return H2OMOJOPipelineModel(javaModel)

    def selectPredictionUDF(self, column):
        java_col = self._java_obj.selectPredictionUDF(column)
        return Column(java_col)

File: r/src/R/ai/h2o/sparkling/ml/models/H2OMOJOPipelineModel.R
source(file.path("R", "H2OMOJOModelBase.R"))

H2OMOJOPipelineModel.createFromMojo <- function(pathToMojo, settings = H2OMOJOSettings.default()) {
  sc <- spark_connection_find()[[1]]
  jmojo <- invoke_static(sc, "ai.h2o.sparkling.ml.models.H2OMOJOPipelineModel", "createFromMojo", pathToMojo, settings$toJavaObject())
  H2OMOJOPipelineModel(jmojo)
}

#' @export H2OMOJOPipelineModel
H2OMOJOPipelineModel <- setRefClass("H2OMOJOPipelineModel", contains = ("H2OAlgorithmMOJOModelBase"), methods = list(
  getWithInternalContributions = function() {
    invoke(.self$jmojo, "getWithInternalContributions")
  },
  getWithPredictionInterval = function() {
    invoke(.self$jmojo, "getWithPredictionInterval")
  },
  getScoringBulkSize = function() {
    invoke(.self$jmojo, "getScoringBulkSize")
  }
))
